README.md

```diff
@@ -1,68 +1,212 @@
+| ISSUANCE_CRED_DEF_POLICY       | y        | EXISTING_OR_NEW                                                | Same as above but this is for credential definitions                                             |
+
+## Testing
+Currently this repository consist of two test types: [**integration**, **unit**] and we are using a combination of `mocha`, `chai` and `sinon` frameworks
+
+### Unit Testing
+Unit tests in this repository are done per service/module or class, and follow the bellow pattern.
+```sh
+│   ├── __tests__
+│   │   ├── auth.test.ts
+│   │   ├── __tests__
+│   │   │   ├── index.test.ts
+│   │   │   └── newConnection.test.ts
+│       ├── __tests__
+│       │   └── index.test.ts
+│   ├── __tests__
+│   │   ├── companyHouseEntity.test.ts
+│   │   ├── credentialSchema.test.ts
+│   │   ├── idpService.test.ts
+│   │   └── veritableCloudagent.test.ts
+│   │   ├── __tests__
+│   │   │   └── index.test.ts
+│   │   ├── __tests__
+│   │   │   ├── companyDetailsV1.test.ts
+│   │   │   └── index.test.ts
+│   ├── __tests__
+│   │   │   └── testIndexedAsyncEventEmitter.ts
+│   │   ├── indexedAsyncEventEmitter.test.ts
+│   │   └── twoWayMap.test.ts
+    │   ├── __tests__
+    │   │   ├── connection.test.ts
+    │   │   └── connection.test.ts.snap
+    │   ├── __tests__
+    │   ├── __tests__
+    │   │   ├── fromInvite.test.ts
+    │   │   ├── fromInvite.test.ts.snap
+    │   │   ├── newInvite.test.ts
+    │   │   ├── newInvite.test.ts.snap
+    │   │   ├── pinSubmission.test.ts
+    │   │   └── pinSubmission.test.ts.snap
+    │   ├── __tests__
+    │   │   ├── queries.test.ts
+    │   │   ├── queries.test.ts.snap
+    │   │   ├── queriesList.test.ts
+    │   │   └── queriesList.test.ts.snap
+``` 
+Unit tests can be executed by running:
+```sh
+npm run test:unit
+```
 
-Seeding data
+### Integration Testing
+Integration tests are placed at the root level of a repository and can be found in `test/` folder along with mock services and helpers and a test environment variables that will be in `test/test.env`.
+```sh
+├── helpers
+│   ├── cloudagent.ts
+│   ├── companyHouse.ts
+│   ├── connection.ts
+│   ├── db.ts
+│   ├── fixtures.ts
+│   ├── logger.ts
+│   ├── routeHelper.ts
+│   └── util.ts
+├── init.ts
+├── integration
+│   ├── newConnection.test.ts
+│   └── pinVerification.test.ts
+├── mocharc.json # mocha configuration
+└── test.env # env variables required for integration tests and mocked services/helpers
+```
 
-```bash
-npm run db:seed
+Integration tests can be run locally by executing the below command
+```sh
+npm run test:integration
 ```
 
-Creating a new seed from template (knex)
+### E2E/UI Testing
+> :This is something that currently is not implemented but it's already on a roadmap so leaving this section for later and if nothing changes we will be using cypress
```

Comment:
Rmeove for now, we'll add it in when we do the e2e tests

Reply:
done